stackslib/src/net/api/gettransaction.rs

```diff
@@ -0,0 +1,197 @@
+    fn try_parse_request(
+        &mut self,
+        preamble: &HttpRequestPreamble,
+        captures: &Captures,
+        query: Option<&str>,
+        _body: &[u8],
+    ) -> Result<HttpRequestContents, Error> {
+        if preamble.get_content_length() != 0 {
+            return Err(Error::DecodeError(
+                "Invalid Http request: expected 0-length body for GetTransaction".to_string(),
+            ));
+        }
+
+        let txid = request::get_txid(captures, "txid")?;
+        self.txid = Some(txid);
+
+        Ok(HttpRequestContents::new().query_string(query))
+    }
+}
+
+impl RPCRequestHandler for RPCGetTransactionRequestHandler {
+    /// Reset internal state
+    fn restart(&mut self) {
+        self.txid = None;
+    }
+
+    /// Make the response
+    fn try_handle_request(
+        &mut self,
+        preamble: HttpRequestPreamble,
+        _contents: HttpRequestContents,
+        node: &mut StacksNodeState,
+    ) -> Result<(HttpResponsePreamble, HttpResponseContents), NetError> {
+        let txid = self
+            .txid
+            .take()
+            .ok_or(NetError::SendError("`txid` no set".into()))?;
+
+        let txinfo_res =
+            node.with_node_state(|_network, _sortdb, chainstate, mempool, _rpc_args| {
+                let index_block_hashes = match chainstate.chainstate_tx_begin() {
+                    Ok((chainstate_tx, clarity_instance)) => {
+                        match chainstate_tx.get_index_block_hashes_from_txid(txid) {
+                            Ok(index_block_hashes) => index_block_hashes,
+                            Err(_) => return Err(NetError::NotFoundError),
+                        }
+                    }
+                    Err(_) => return Err(NetError::NotFoundError),
+                };
+
+                // search for the first matching tx with valid index_block_hash
+                for index_block_hash in index_block_hashes {
+                    match chainstate
+                        .nakamoto_blocks_db()
+                        .get_nakamoto_block(&index_block_hash)
+                    {
+                        Ok(nakamoto_block) => {
+                            for tx in nakamoto_block.unwrap().0.txs {
+                                if tx.txid() == txid {
+                                    return Ok(TransactionResponse {
+                                        block_hash: index_block_hash.to_hex(),
+                                        tx: to_hex(&tx.serialize_to_vec()),
+                                    });
+                                }
+                            }
+                        }
+                        Err(_) => return Err(NetError::NotFoundError),
+                    }
+                }
+
+                return Err(NetError::NotFoundError);
+            });
+
+        if txinfo_res.is_err() {
+            return StacksHttpResponse::new_error(
```

Comment:
How do you know it's solely due to the transaction not being present?  What if it's a DB-related error?